src/main/java/org/datadog/jmxfetch/util/CustomLogger.java

```diff
@@ -1,123 +1,192 @@
+                    fileHandler.setFormatter(formatter);
+                    // note: fileHandler defaults to Level.ALL, so no need to set its log level
+                } catch (Exception e) {
+                    fileHandler = null;
+                    log.error("can't open the file handler:", e);
+                }
+            } else if (isStdErr(logLocation)) {
+                // console handler sending on stderr
+                // note that ConsoleHandler is sending on System.err
+                stderrHandler = new ConsoleHandler();
+                stderrHandler.setFormatter(formatter);
+                stderrHandler.setLevel(julLevel);
             }
         }
 
-        // replace default appender with the correct layout
-        LoggerConfig loggerConfig = config.getLoggerConfig(LogManager.ROOT_LOGGER_NAME);
-        loggerConfig.removeAppender(target);
+        // always have a console handler sending the logs to stdout
+        stdoutHandler = new StdoutConsoleHandler();
+        stdoutHandler.setFormatter(formatter);
+        stdoutHandler.setLevel(julLevel);
 
-        Appender appender = ConsoleAppender.newBuilder()
-                    .setConfiguration(config)
-                    .withName(target)
-                    .withLayout(layout)
-                    .build();
-        appender.start();
+        logger.setLevel(julLevel);
 
-        loggerConfig.addAppender(appender, null, null);
-        loggerConfig.setLevel(level);
+        // set our configured handlers
+        if (fileHandler != null) {
+            logger.addHandler(fileHandler);
+        }
+        if (stdoutHandler != null) { // always non-null but doesn't cost much
+            logger.addHandler(stdoutHandler);
+        }
+        if (stderrHandler != null) {
+            logger.addHandler(stderrHandler);
+        }
+
+        // filter all other log handlers
+        for (Enumeration<String> logNames = manager.getLoggerNames() ;
+                logNames.hasMoreElements() ; ) {
+            String logName = logNames.nextElement();
+
+            for (Handler handler : manager.getLogger(logName).getHandlers()) {
+                if (handler != null) {
+
+                    handler.setFilter(new Filter() {
+                        @Override
+                        public boolean isLoggable(LogRecord record) {
+                            return false;
+                        }
+                    });
+                }
+            }
+        }
+
+        // set our jmxfetch handlers
+        for (Handler handler : logger.getHandlers()) {
+            handler.setFilter(new Filter() {
+                @Override
+                public boolean isLoggable(LogRecord record) {
+                    String sourceClass = record.getSourceClassName();
+                    return sourceClass.startsWith("org.datadog.jmxfetch");
+                }
+            });
+        }
```

Comment:
I think we can achieve this by creating our logger named `"org.datadog.jmxfetch"`. Since all the JMXFetch classes use the `@Slf4j` annotation, they use a `log`ger that's a child of `org.datadog.jmxfetch` (https://projectlombok.org/api/lombok/extern/slf4j/Slf4j.html)

Reply:
I've addressed this, resolving